File: lesson_2/2_java_example/Student.java
package example;

import example.Course;
import java.util.ArrayList;

public class Student {
  private String name;
  private ArrayList<Course> courses;

  public Student(String name) {
    this.name = name;
    this.courses = new ArrayList<Course>();
  }

  public String getName() {
    return name;
  }

  public ArrayList<Course> getCourses() {
    return courses;
  }

  public void addCourse(Course course) {
    courses.add(course);
  }

  public void removeCourse(Course course) {
    courses.remove(course);
  }

  public String toString() {
    var stringBuilder = new StringBuilder();

    stringBuilder.append("Name: " + name + "\tCourses: ");

    for (var course : courses) {
      stringBuilder.append("\t" + course.toString() + "\n");
    }

    return stringBuilder.toString();
  }
}

File: lesson_2/3_python_example/student.py
# Define a class consisting of a student name and a list of courses.
class Student:
  def __init__(self, name, courses):
    self.name = name
    self.courses = courses

  def __str__(self):
    return f'Name: {self.name}\tCourses: {self.courses}'

  def add_course(self, course):
    self.courses.append(course)

  def calculate_average_grade(self):
    total_grade = 0
    for course in self.courses:
      total_grade += course.grade
    return total_grade / len(self.courses)
